### PR TITLE
Dịch chương 42

### LINKED ISSUE #290: Dịch chương 42

### PR DESCRIPTION
**Pull request by naml3i**
## Nếu bạn tạo Pull Request để dịch, vui lòng hoàn thành các thông tin dưới đây.

* [x] Đánh dấu [x] (không có dấu cách quanh x) thể hiện dấu tick vào các câu dưới đây cho các câu trả lời 'Có' hoặc 'Đúng'

* [x] Pull Request này được link tới một issue? Đánh số issue đó *ngay sau* dấu `#`: close #290 

* [x] Bạn có kiến thức về Học Máy không? Chúng tôi ưu tiên những người có trước kiến thức về Học Máy để việc review được thuận tiện hơn.

* [x] Bạn đã đọc [Bảng thuật ngữ](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/glossary.md) chưa? Bạn cần dịch các thuật ngữ theo bảng thuật ngữ chung.

* [x] Tất cả các thuật ngữ trong chương này đã có trong [Bảng thuật ngữ](https://github.com/aivivn/Machine-Learning-Yearning-Vietnamese-Translation/blob/master/glossary.md) chưa? Nếu chưa, bạn có thể tạo một issue hoặc pull request mới để bổ sung nghĩa của từ đó vào Bảng thuật ngữ.

* [x] Chọn milestone cho chương này ở cột bên phải.

* [x] Pull request này đã sẵn sàng cho việc review chưa? Nếu đã sẵn sàng, bạn vui lòng gán label 'review wanted' cho pull request này. Nếu chưa, bạn vui lòng thêm cụm "[WIP]" (work in progress) vào trước tiêu đề của pull request.

## Lưu ý 
**Chỉ bắt đầu review khi mọi mục trên đây đều đã được tick.**

### CODE REVIEW

**Comment by naml3i** [comment]
- "Data mismatch", hoặc "mismatched data" hiện thời mình chọn từ "không tương đồng", VD cụ thể thế nào là không tương đồng thì như ở trong chương, hoặc ở chương trước (VD như xây dựng app nhận dạng mèo trên mobile nhưng phần lớn ảnh training lấy được từ crawling web nên chất lượng ảnh cao hơn hẳn chất lượng ảnh của hệ thống ngoài đời thật (chụp bằng mobile -- rung, mờ, ánh sáng yếu, vv....). 
Cái đó là "không tương đồng" (hoàn toàn khác với mislabeled). Mời mọi người cho ý kiến về cách dịch "mismatch". Ngoài ra mình nghĩ được vài option là không tương đồng/tương xứng/tương thích/tương đương/cùng tính chất/.... Ban đầu chọn không tương xứng rồi chuyển thành tương đồng.

- Thuật ngữ `training dev set` không biết có được tính là đã có trong bảng thuật ngữ chưa (mặc dù cả training và dev set đều có rồi). Khái niệm này lần đầu được nhắc đến ở vài chương trước chương này, và có lẽ cần được thống nhất giữa người dịch các chương.

- Một số nơi thì dùng training-dev set (có dấu gạch), một số dùng training/dev set, trong sách này lại viết không có gạch nối gì. Mình prefer có gạch nối để thể hiện đây là một subset đặc biệt để validate riêng cái training set (không phải dev set và test set chính thống): "tập huấn luyện-phát triển" (hay "phát triển-huấn luyện"?). @tiepvupsu cân nhắc có cần mở pull request không nhé.

- [Ví dụ về cách viết](https://yashuseth.blog/2018/03/20/what-to-do-when-we-have-mismatched-training-and-validation-set/) `training-dev set` (có gạch nối)

**Comment by tiepvupsu** [comment]
> * ồi). Khái niệm này lần đầu được nhắc đến ở vài chương trước chương này, và có lẽ cần được thống nhất giữa người dịch các chương.
> * Một số nơi thì dùng training-dev set (có dấu gạch), một s

Xin lỗi bạn mình   missed comment này. Mình có để lại comment trong #305. Mình khá thích cụm "dữ liệu không tương đồng", cùm này chuẩn hơn "dữ liệu không tương đương".

File: chapters/ch42.md [hunk 1]
```diff
@@ -1,35 +1,37 @@
 > # 42. Addressing data mismatch
 
---> _replace THIS LINE by your translation for the above line_
+# 42. Xử lý dữ liệu không tương đồng
```
Comment (tiepvupsu) [inline chapters/ch42.md:3]: 👍

File: chapters/ch42.md [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
 
 > Suppose you have developed a speech recognition system that does very well on the training set and on the training dev set. However, it does poorly on your dev set: You have a data mismatch problem. What can you do?
 
---> _replace THIS LINE by your translation for the above line_
+Giả sử bạn đã phát triển một hệ thống nhận dạng giọng nói hoạt động rất tốt  trên tập huấn luyện và trên tập huấn luyện - phát triển. Tuy nhiên, hệ thống đó lại hoạt động kém trên tập phát triển: Bạn có vấn đề về dữ liệu không tương đồng. Bạn có thể làm gì?
```
Comment (tiepvupsu) [inline chapters/ch42.md]: * [ ] Hai dấu cách giữa "tốt" và "trên tập"

Comment (tiepvupsu) [reply chapters/ch42.md]: Cụm "training dev set" mình đề xuất dịch là "tập phát triển trích từ huấn luyện".

File: chapters/ch42.md [hunk 3]
```diff
@@ -1,35 +1,37 @@
...
 
---> _replace THIS LINE by your translation for the above line_
+Ví dụ, giả sử bạn thực hiện phân tích lỗi nhận dạng giọng nói trên tập phát triển: Bạn duyệt qua 100 mẫu một cách thủ công và cố gắng hiểu xem thuật toán đang phạm lỗi ở những đâu. Bạn phát hiện rằng hệ thống của bạn thực thi kém vì hầu hết những đoạn âm thanh trong tập phát triển được thu trong ô tô, trong khi hầu hết các ví dụ để huấn luyện được thu trên một nền yên tĩnh. Tiếng ồn từ động cơ và đường phố làm ảnh hưởng nghiêm trọng đến chất lượng của hệ thống của bạn. Trong trường hợp này, bạn có thể cố gắng thu thập thêm dữ liệu huấn luyện bao gồm những đoạn âm thanh được thu trong ô tô. Mục đích của việc phân tích lỗi là để hiểu những khác biệt đáng kể giữa tập huấn luyện và tập phát triển, vốn là nguyên nhân dẫn đến dữ liệu không tương đồng.
```
Comment (tiepvupsu) [inline chapters/ch42.md]: * Chỗ "phạm lỗi" nghe hơi nghiêm trọng, "có lỗi" thì thế nào?
* "thực thi" -> "hoạt động"?
* "car" được dịch là "xe hơi" ở vài chương trước, nên thống nhất chọn một cách. Mình đề xuất là "xe hơi"
* "nền yên tĩnh" -> "môi trường yên tĩnh"?

File: chapters/ch42.md [hunk 4]
```diff
@@ -1,35 +1,37 @@
...
 
---> _replace THIS LINE by your translation for the above line_
+Nếu tập huấn luyện và tập huấn luyện-phát triển của bạn chứa những đoạn audio thu trong ô tô, bạn nên kiểm tra kĩ lưỡng chất lượng của hệ thống của bạn trên tập con dữ liệu này. Nếu hệ thống hoạt động tốt với dữ liệu ô tô trong tập huấn luyện nhưng không tốt với dữ liệu ô tô trong tập huấn luyện-phát triển, điều đó càng khẳng định giả thuyết rằng thu thập thêm dữ liệu ô tô sẽ có ích. Đó là lí do tại sao chúng ta thảo luận trong chương trước về khả năng thêm một số dữ liệu lấy từ tập phát triển/tập kiểm tra với cùng phân phối vào trong dữ liệu huấn luyện của bạn. Làm như vậy cho phép bạn so sánh chất lượng [hệ thống] trên tập huấn luyện so với trên tập phát triển/kiểm tra.
```
Comment (tiepvupsu) [inline chapters/ch42.md]: "audio" -> "âm thanh"

File: chapters/ch42.md [hunk 5]
```diff
@@ -1,35 +1,37 @@
...
 
 
 > **FOOTNOTE:**
 
 > [14] There is also some research on "domain adaptation"—how to train an algorithm on one distribution and have it generalize to a different distribution. These methods are typically applicable only in special types of problems and are much less widely used than the ideas described in this chapter.
 
---> _replace THIS LINE by your translation for the above line_
+**GHI CHÚ**
+
+[14] Cũng có một số nghiên cứu về "thích nghi theo miền" -- làm sao để huấn luyện một giải thuật trên một phân phối và để nó tổng quát hóa trên một phân phối khác. Những phương pháp này thường chỉ ứng dụng được với một số loại vấn đề đặc biệt, và ít được sử dụng hơn nhiều so với những ý tưởng đã được trình bày trong chương này.
```
Comment (tiepvupsu) [inline chapters/ch42.md]: * "domain adaption" đã có trong glossary, nó là "thích ứng miền".
* Từ "algorithm" có thể dịch là "thuật toán" hoặc" giải thuật", nhưng cần nhất quán. Từ đầu sách nó được dịch là "thuật toán" nên mình đề xuất bạn dịch lại là "thuật toán".
* Từ "problem" có nhiều nghĩa, vừa là "vấn đề", vừa là "bài toán". Trong trường hợp này mình nghĩ "bài toán" hợp lý hơn.

File: chapters/ch42.md (hunk 3 above)
Comment (naml3i) [reply chapters/ch42.md]: 👍 

- --> "Mắc lỗi"? (thuật toán "có lỗi" nghe giống như faulty, là một intrinsic characteristic vậy. "Mắc" nghe giống vấp phải lỗi, gây ra lỗi trong quá trình hoạt động hơn. Một thuật toán "có lỗi" trong bản chất vẫn có xác suất "may" khi thực hiện không mắc lỗi nào.)

- "Hoạt động" nghe tự nhiên hơn ổn hơn, hơn nữa "thực thi" thường phải có object chứ không đi một mình như một intransitive verb (như "hoạt động" được --> OK thay bằng "hoạt động".

- "car" mình sẽ update. Cái này không quan trọng, nếu đã có dịch trước rồi thì cứ theo thôi ;) 

- Đúng là "môi trường yên tĩnh" thì có vẻ quen tai hơn. Thực ra nếu chi li trong context thu âm thì có thể "background" có sắc thái riêng. Dù sao cũng không quan trọng nên mình sẽ đổi thành "môi trường"

File: chapters/ch42.md (hunk 2 above)
Comment (naml3i) [reply chapters/ch42.md]: sẽ sửa thành "tập phát triển huấn luyện" theo như vote ở PR #305

**Review by tiepvupsu — APPROVED** [review]
Chương này mình thấy rất ổn rồi. Cảm ơn bạn nhiều.
